primitives/session/src/lib.rs

```diff
@@ -130,3 +130,29 @@ where
 
 	Ok(())
 }
+
+/// A trait to be notified when the session changes
+pub trait SessionChangeListener {
+	fn on_session_change(session_index: SessionIndex);
+}
+
+#[impl_trait_for_tuples::impl_for_tuples(1, 30)]
+#[tuple_types_custom_trait_bound(SessionChangeListener)]
```

Comment:
```suggestion
```

Why do we need this?

Reply:
If you look at the documentation of `impl_trait_for_tuples`, I am sure this can work with the most simple syntax, which means you can just put this on top of the trait and it will work out of the box.